docs/developer-guide/plugins.md

```diff
@@ -52,7 +63,31 @@ ruleFunction.ruleName = ruleName;
 ruleFunction.messages = messages;
 ruleFunction.meta = meta;
 
-module.exports = stylelint.createPlugin(ruleName, ruleFunction);
+export default stylelint.createPlugin(ruleName, ruleFunction);
+```
+
+> [!IMPORTANT]
+> A CommonJS plugin also works, but such a plugin may be unsupported in the future.
+> We recommend writing an ES module plugin if compatibility doesn't matter.
```

Comment:
```suggestion
```

Let's take this out for now, and add something about commonjs plugins being deprecated in a follow-up issue.